windows/ReactTestApp/MainPage.cpp

```diff
@@ -118,4 +121,52 @@ namespace winrt::ReactTestApp::implementation
         TitleBar().Height(sender.Height());
     }
 
+    void MainPage::WriteInitialProperties(
+        std::optional<std::map<std::string, std::any>> const &initialProps)
+    {
+        if (initialProps.has_value()) {
+            ReactRootView().InitialProps([initialProps](IJSValueWriter const &writer) {
+                writer.WriteObjectBegin();
+                for (auto &&property : initialProps.value()) {
+                    auto &value = property.second;
+                    writer.WritePropertyName(to_hstring(property.first));
+                    WritePropertyValue(value, writer);
+                }
+                writer.WriteObjectEnd();
+            });
+        }
+    }
+
+    void WritePropertyValue(std::any const &propertyValue, IJSValueWriter const &writer)
+    {
+        if (propertyValue.type() == typeid(boolean)) {
+            writer.WriteBoolean(std::any_cast<boolean>(propertyValue));
+        } else if (propertyValue.type() == typeid(std::int64_t)) {
+            writer.WriteInt64(std::any_cast<std::int64_t>(propertyValue));
+        } else if (propertyValue.type() == typeid(std::uint64_t)) {
+            writer.WriteInt64(std::any_cast<std::uint64_t>(propertyValue));
+        } else if (propertyValue.type() == typeid(double)) {
+            writer.WriteDouble(std::any_cast<double>(propertyValue));
+        } else if (propertyValue.type() == typeid(std::nullopt)) {
+            writer.WriteNull();
+        } else if (propertyValue.type() == typeid(std::basic_string<char>)) {
```

Comment:
`std::string` is the same as `std::basic_string<char>` but is more legible:
```suggestion
        } else if (propertyValue.type() == typeid(std::string)) {
```